src/identity/client.rs

```diff
@@ -70,6 +74,41 @@ impl FullId {
     }
 }
 
+impl From<BlsSecretKey> for FullId {
+    fn from(bls_sk: BlsSecretKey) -> Self {
+        let public = bls_sk.public_key();
+        let keypair = Keypair::Bls(BlsKeypair {
+            secret: SerdeSecret(bls_sk),
+            public,
+        });
+        let public_key = keypair.public_key();
```

Comment:
I don't think we need this line, we can just reuse the `public` variable.

Reply:
I believe public is of type `threshold_crypto::PublicKey` and `public_key` is of type `safe_nd::PublicKey`. So we'll need both of them.